src/SdkCommon/ClientRuntime/ClientRuntime/ExtensibleEnum.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+namespace Microsoft.Rest
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+
+    /// <summary>
+    /// Abstract base class for types that act like enums, but can be extended with arbitrary string values.
+    /// </summary>
+    public abstract class ExtensibleEnum<T> : IEquatable<T> where T : ExtensibleEnum<T>
```

Comment:
let `ExtensibleEnum` take another generic argument (I'll call it `TVal`) that determines the type of stuff that the enum is about

Reply:
Do you have a particular scenario in mind? Attaching labels to a fixed set of integers, for example...?

Reply:
@olydis I am trying to generalize this as you suggested but the problem is when I try to create an `ExtensibleEnum<T,V>` type I need to provide the name of the enum as well as its underlying (primitive) type. I don't think we capture this information in `EnumType.cs` so there really isn't an easy way to figure this out.
Any ideas?

Reply:
@brjohnstmsft exactly, supporting non-string enums is one of the bigger asks we'll probably tackle some time soon (appears in some data-plane stuff we saw coming in) 🙂

Reply:
@dsgouda not sure I understand, pretty sure this is expressible - what's the error you get when doing what?